### PR TITLE
feat: header and navigation

### PR DESCRIPTION
**Pull request by pfulton**
Styles the site-wide header and navigation.

URL for testing:

- https://feat--header--design-website--adobe.hlx3.page/

## Description
* moves the `setBodyColor` function into its own file for better reuse.
* leans on the `light-text`/`dark-text` functionality that was built into the homepage, and makes it available for site-wide usage in the header
* colors the logo svg, "tagline" svg, navigation elements, and navigation icons according to values found in the `colorMap`: light text is applied to dark backgrounds and vice versa.
* builds the small-screen navigation experience according to the Xd mockups.
* builds the medium+ navigation experience according to the Xd mockups.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: scripts/page-background.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  blendedBackground.classList.add('blended-background');
+  blendedBackground.style.setProperty('--blended-background-color', pageBgColor);
+
+  document.body.append(blendedBackground);
+  document.documentElement.style.setProperty('--header-color', pageBgColor);
+}
+
+if (window.location.pathname !== '/') {
+  setPageBackgroundColor();
+}
```
Comment (pfulton) [inline scripts/page-background.js:18]: @cjgammon - a little bit of an explainer here: for all pages (_except_ the homepage), [I'm appending a `div`](https://github.com/adobe/design-website/pull/30/files#diff-37f6471b306924bbb531d040e7ab55df5f83511246a411d22947462323516afaR164-R171) with a solid-to-transparent gradient. This changes based on the page's set background color (via the cms).

**Comment by cjgammon** [comment]
This seems fine, one issue I noticed is the header's background color seems to fade in the beginning after page load.  This seems to be because there is a transition on its background color and it is being set after the content loads.  I think a better approach might be to make it transparent and then have padding at the top of the slides to account for the header height?  This might help prevent the header from jumping too.

**Comment by pfulton** [comment]
> This seems fine, one issue I noticed is the header's background color seems to fade in the beginning after page load. This seems to be because there is a transition on its background color and it is being set after the content loads. I think a better approach might be to make it transparent and then have padding at the top of the slides to account for the header height? This might help prevent the header from jumping too.

Yeah, I'd noticed that, too. I'll take a shot at addressing that with your suggestions.

**Comment by cjgammon** [comment]
One other quick note but this may be unrelated to the PR, I'm not sure, the teams page has a white bg on this branch.  Could be that a variable isn't set for it though?
Screen Shot 2022-02-07 at 9 51 57 AM

**Comment by pfulton** [comment]
>One other quick note but this may be unrelated to the PR, I'm not sure, the teams page has a white bg on this branch. Could be that a variable isn't set for it though?

Thanks for the heads up. This page didn't have its color set on the Word doc. Should be fixed now (it's `#E91D25` - red)

**Pull request merged**